### ISSUE TITLE
Implement Search Feature in Filter Menu

### ISSUE THREAD

**Issue by Bonapara** [twentyhq/twenty#4367]
# Current Behavior

![CleanShot 2024-03-07 at 18 19 14](https://github.com/twentyhq/twenty/assets/19412894/53354048-1849-4d70-b6c8-07e587930ecc)

# Desired Behavior

The filter menu currently lacks a 'Search' function. To optimize keyboard navigation and handle numerous fields effectively, we propose the addition of a search feature at the top

![CleanShot 2024-03-07 at 18 23 48](https://github.com/twentyhq/twenty/assets/19412894/af1e516b-b444-445a-bdca-d0ceb2053667)

# Figma

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=2979-194754&mode=design&t=LORfESmPs8dJ3RY2-11

**Comment by ahmadjafari86** [comment]
Hi @Bonapara 
I'd like to work on this issue.

**Comment by Bonapara** [comment]
Thanks @ahmadjafari86!

**Comment by ahmadjafari86** [comment]
hey @Bonapara can you add me to the Figma team for this project?

**Comment by ahmadjafari86** [comment]
Hey @Bonapara, just a heads up, I've unassigned myself from this task. I'm having trouble running the server on my local system.

**Comment by Bonapara** [comment]
Understood, thanks for your message @ahmadjafari86!

**Comment by rhlshah** [comment]
I would like to take this up!

**Comment by Bonapara** [comment]
Just assigned you, thanks @rhlshah :)

**Comment by rathodvinod5** [comment]
@Bonapara, could you please assign this task to me, because this task is similar to #4368, so it will be just a matter of copy paste for me.

**Comment by Bonapara** [comment]
Sure, thanks @rathodvinod5. Unassigning @rhlshah as it's been a while!
